packages/next/src/server/app-render/action-handler.ts

```diff
@@ -181,35 +177,33 @@ async function createForwardedActionResponse(
   const fetchUrl = new URL(`${origin}${basePath}${workerPathname}`)
 
   try {
-    let readableStream: ReadableStream<Uint8Array> | undefined
-    if (process.env.NEXT_RUNTIME === 'edge') {
-      const webRequest = req as unknown as WebNextRequest
-      if (!webRequest.body) {
-        throw new Error('invariant: Missing request body.')
+    let body: BodyInit | AsyncIterable<any> | undefined
+    if (
+      // The type check here ensures that `req` is correctly typed, and the
+      // environment variable check provides dead code elimination.
+      process.env.NEXT_RUNTIME === 'edge' &&
+      isWebNextRequest(req)
```

Comment:
maybe a dumb question, but will `isWebNextRequest(req)` not properly DCE if we don't duplicate & inline the env check? 

I ran it through Terser and it seemed to work as I would've expected but maybe you observed differently

Reply:
I'll double check, but my experiments showed that functions weren't "constant", therefore couldn't be relied on for DCE. If it works, I'll do that instead 😅

Reply:
After reviewing, looks like after turning off the `process.env.` checks prevented the DCE, and the bundled edge page files contained a string that I added just to the node branch.